public/public.php

```diff
@@ -906,10 +909,8 @@ public function get_thumbnail_source_string( int $post_id = 0 ): string {
 			$thumb = get_post( $post_id );
 
 			// don’t show caption for own image if admin choose not to do so
-			if ( $this->is_standard_source( 'exclude' ) ) {
-				if ( self::use_standard_source( $id ) ) {
-					return '';
-				}
+			if ( Standard_Source::hide_standard_source_for_image( $id ) ) {
+				return '';
 			}
```

Comment:
**Suggestion:** It seems that you are using the `Standard_Source` class methods statically. If `Standard_Source` is not designed as a utility class, consider instantiating it and using its methods as instance methods. This can make your code more testable and maintainable. [best practice]
```suggestion
   $standardSource = new Standard_Source();
   if ( $standardSource->hide_standard_source_for_image( $id ) ) {
       return '';
   }
```